src/serac/physics/utilities/variational_form/tuple_arithmetic.hpp

```diff
@@ -0,0 +1,214 @@
+#pragma once
+
+#include <tuple>
+
+#include "tensor.hpp"
+
+namespace serac {
+
+namespace detail {
+
+/**
+ * @brief Trait for checking if a type is a @p std::tuple
+ */
+template <typename T>
+struct is_tuple {
+  // FIXME: Should we use std::false_type/std::true_type
+  static constexpr bool value = false;
+};
+/// @overload
+template <typename... T>
+struct is_tuple<std::tuple<T...> > {
+  static constexpr bool value = true;
+};
+
+/**
+ * @brief Trait for checking if a type if a @p std::tuple containting only @p std::tuple
+ */
+template <typename T>
+struct is_tuple_of_tuples {
+  static constexpr bool value = false;
+};
+/// @overload
+template <typename... T>
+struct is_tuple_of_tuples<std::tuple<T...> > {
+  static constexpr bool value = (is_tuple<T>::value && ...);
+};
+
+/////////////////////////////////////////////////
+
+template <typename... S, typename... T, int... I>
```

Comment:
I think clearer variables names help with readability since I'm not great at templates. This is also in a `detail` namespace. Perhaps we can move it to an issue to resolve later?